### PR TITLE
[Inductor] Add MPS device op overrides

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #143893
* __->__ #143892

Mostly dummy interface as MPS backend currently limited to a single device

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/codegen/common.py [hunk 1]
```diff
@@ -434,7 +434,10 @@ def get_device_op_overrides(device: str):
     assert isinstance(device, str)
 
     if not device_op_overrides_dict.keys():
-        from . import cpu_device_op_overrides  # noqa: F401
+        from . import (  # noqa: F401  # noqa: F401
```
Comment (jansel) [inline torch/_inductor/codegen/common.py:437]: Does double noqa do anything extra?

File: torch/_inductor/codegen/mps_device_op_overrides.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# mypy: allow-untyped-defs
```
Comment (jansel) [inline torch/_inductor/codegen/mps_device_op_overrides.py:1]: Remove this and add types

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_inductor/codegen/common.py (hunk 1 above)
Comment (malfet) [reply torch/_inductor/codegen/common.py]: Nope, just linter being linter.. Fixed.

File: torch/_inductor/codegen/mps_device_op_overrides.py (hunk 2 above)
Comment (malfet) [reply torch/_inductor/codegen/mps_device_op_overrides.py]: Done

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**